terraform/log.go

```diff
@@ -0,0 +1,65 @@
+package terraform
+
+import (
+	"regexp"
+	"strings"
+	"time"
+
+	"github.com/gruntwork-io/go-commons/errors"
+	"github.com/gruntwork-io/terragrunt/pkg/log"
+	"github.com/gruntwork-io/terragrunt/pkg/log/writer"
+)
+
+const (
+	// logTimestampFormat is TF_LOG timestamp format
+	logTimestampFormat = "2006-01-02T15:04:05.000-0700"
+)
+
+var (
+	// tfLogTimeLevelMsgReg is a regular expression that matches TF_LOG output, example output:
+	//
+	// 2024-09-08T13:44:31.229+0300 [DEBUG] using github.com/zclconf/go-cty v1.14.3
+	// 2024-09-08T13:44:31.229+0300 [INFO]  Go runtime version: go1.22.1
+	tfLogTimeLevelMsgReg = regexp.MustCompile(`(?i)(^\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}\.\d{3}[-+]\d{4})\s*\[(trace|debug|warn|info|error)\]\s*(.+\S)$`)
```

Comment:
Maybe at a minimum, we should have a test that only parses TF_LOG as confirmation for the current version of OpenTofu that we're testing if we don't already have that.

Reply:
> Is this guaranteed? What happens if OpenTofu changes their formatting?

Nothing bad will happen, just the time and log level will be added to the message instead of the highlighted columns.

> Maybe at a minimum, we should have a test that only parses TF_LOG as confirmation for the current version of OpenTofu that we're testing if we don't already have that.

Good idea.